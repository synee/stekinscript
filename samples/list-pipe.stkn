ls: [5, 1, 6, 7, 2, 9, 4, 3, 8, 0]

console.log(ls |: $ * $)
console.log(ls |: $ % 2 = 0)
console.log(ls |? $ % 2 = 0)

func head(list, n)
    return list |? $i < n

func after(list, n)
    return list |? $i >= n

console.log(head(ls, 4))
console.log(after(ls, 4))
console.log(ls |: $ * $ % 7 |? $ < 4)

console.log('--- x ---')
console.log(ls |: (['sd', 'a', 'ab', 'abc'] |? $.length % 2 = 0) ++ [$])

func getIndex(i)
    return (): i
ils: ls |: getIndex($i)
console.log(ils |: $())

func getKey(k)
    return (): k
kls: ls |: getKey($k)
console.log(kls |: $())

console.log(ls |:
    sqr: $ * $
    if sqr % 3 = 0
        $r.push(sqr)
    else
        $r.push(-sqr)
)

console.log(ls |:
    sqr: $ * $
    if sqr % 3 = 0
        $r.push(sqr)
)

console.log(ls |:
    sqr: $ * $
    # oh lost push
)

console.log('End')
