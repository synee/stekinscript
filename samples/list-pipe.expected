[ 25, 1, 36, 49, 4, 81, 16, 9, 64, 0 ]
[ false, false, true, false, true, false, true, false, true, true ]
[ 6, 2, 4, 8, 0 ]
[ 5, 1, 6, 7 ]
[ 2, 9, 4, 3, 8, 0 ]
[ 1, 1, 0, 2, 2, 1, 0 ]
--- x ---
[ [ 'sd', 'ab', 5 ],
  [ 'sd', 'ab', 1 ],
  [ 'sd', 'ab', 6 ],
  [ 'sd', 'ab', 7 ],
  [ 'sd', 'ab', 2 ],
  [ 'sd', 'ab', 9 ],
  [ 'sd', 'ab', 4 ],
  [ 'sd', 'ab', 3 ],
  [ 'sd', 'ab', 8 ],
  [ 'sd', 'ab', 0 ] ]
[ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 ]
[ '0', '1', '2', '3', '4', '5', '6', '7', '8', '9' ]
[ -25, -1, 36, -49, -4, 81, -16, 9, -64, 0 ]
[ 36, 81, 9, 0 ]
[]
End
